axelrod/strategies/axelrod_first.py

```diff
@@ -73,107 +73,181 @@ def strategy(self, opponent: Player) -> Action:
+    I.E. The player aims to maximise the expected value of being in each state
+    weighted by the number of times we expect to be in that state.
+
+    On the second page of the appendix, figure 4 (page 390) supposedly
+    identifies an expression for EV_TOT however it is not clear how some of the
+    steps are carried out. To the best guess, it seems like an asymptotic
+    argument is being used. Furthermore, a specific term is made to disappear in
+    the case of T - R = P - S (which is not the case for the standard
+    (R, P, S, T) = (3, 1, 0, 5)):
+
+    > "Where (t - r) = (p - s), EV_TOT will be a function of alpha, beta, t, r,
+    p, s and N are known and V which is unknown.
+
+    V is the total number of cooperations of the player S (this is noted earlier
+    in the abstract) and as such the final expression (with only V as unknown)
+    can be used to decide if V should indicate that S always cooperates or not.
+
+    Given the lack of usable details in this paper, the following interpretation
+    is used to implement this strategy:
+
+    1. On any given turn, the strategy will estimate alpha = P(C_o | C_s) and
+    beta = P(C_o | D_s).
+    2. The stragy will calculate the expected utility of always playing C OR
+    always playing D against the estimage probabilities. This corresponds to:
+
+        a. In the case of the player always cooperating:
+
+           P_CC = alpha and P_CD = 1 - alpha
+
+        b. In the case of the player always defecting:
+
+           P_DC = beta and P_DD = 1 - beta
+
+
+    Using this we have:
+
+        E_C = alpha R + (1 - alpha) S
+        E_D = beta T + (1 - beta) P
+
+    Thus at every turn, the strategy will calculate those two values and
+    cooperate if E_C > E_D and will defect if E_C < E_D.
+
+    In the case of E_C = E_D, the player will alternate from their previous
+    move. This is based on specific sentence from Axelrod's original paper:
+
+    > "Under certain circumstances, DOWNING will even determine that the best
+    > strategy is to alternate cooperation and defection."
+
+    One final important point is the early game behaviour of the strategy. It
+    has been noted that this strategy was implemented in a way that assumed that
+    alpha and beta were both 1/2:
+
+    > "Initially, they are both assumed to be .5, which amounts to the
+    > pessimistic assumption that the other player is not responsive."
+
+    Thus, the player opens with a defection in the first two rounds. Note that
+    from the Axelrod publications alone there is nothing to indicate defections
+    on the first two rounds, although a defection in the opening round is clear.
+    However there is a presentation available at
+    http://www.sci.brooklyn.cuny.edu/~sklar/teaching/f05/alife/notes/azhar-ipd-Oct19th.pdf
+    That clearly states that Downing defected in the first two rounds, thus this
+    is assumed to be the behaviour.
+
+    Note that response to the first round allows us to estimate
```

Comment:
I now understand these two sentences but I had to look at the code. I am happy to suggest a change but I have one question before I do.

> we will use the opening play of the player to estimate alpha = P(C_o | C_s)

Why we don't assume that `P(C_o | D_s) = P(C_o | C_s) = 1/2` as stated above?